Graph/Basic Graph Traversal/1462.py

```diff
@@ -0,0 +1,55 @@
+class Solution:
+    def checkIfPrerequisite(self, numCourses: int, prerequisites: List[List[int]], queries: List[List[int]]) -> List[bool]:
+        ans = []
+        self.graph = defaultdict(list)
+
+        # Build an adjacency list to represent graph.
+        for a, b in prerequisites:
+            self.graph[a].append(b)
+
+        for u, v in queries:
+            ans.append(self.isPrerequisites(u, v))
+
+        return ans
+
+    # BFS Solution.
+    def isPrerequisites(self, start: int, end: int) -> bool:
```

Comment:
This BFS implementation is good! Hopefully now you can compare and contrast these three types of BFS problems:

1. Grids
2. Trees
3. Implicit graph problems modeled into a graph (in the form of adjacency list or adjacency matrix)